src/utils/wallet.js

```diff
@@ -438,94 +521,134 @@ class Wallet {
         }
     }
 
-    async signatureFor(accountId) {
-        const blockNumber = String((await this.connection.provider.status()).sync_info.latest_block_height);
-        const signed = await this.inMemorySigner.signMessage(Buffer.from(blockNumber), accountId, NETWORK_ID);
+    async signatureFor(account) {
+        const { accountId } = account
+        const blockNumber = String((await account.connection.provider.status()).sync_info.latest_block_height);
+        const signer = account.inMemorySigner || account.connection.signer
+        const signed = await signer.signMessage(Buffer.from(blockNumber), accountId, NETWORK_ID);
         const blockNumberSignature = Buffer.from(signed.signature).toString('base64');
         return { blockNumber, blockNumberSignature };
     }
 
     async postSignedJson(path, options) {
+        // if there's a tempTwoFactorAccount (recovery with 2fa) use that account
         return await sendJson('POST', ACCOUNT_HELPER_URL + path, {
             ...options,
-            ...(await this.signatureFor(this.accountId))
+            ...(await this.signatureFor(this.tempTwoFactorAccount ? this.tempTwoFactorAccount : this))
         });
     }
 
-    async initializeRecoveryMethod(accountId, method) {
-        return await this.postSignedJson('/account/initializeRecoveryMethod', {
-            accountId,
-            method
-        });
+    async initializeRecoveryMethod(accountId, method, isNew) {
+        if (isNew) {
```

Comment:
contract helper can already check if account is new, not sure why have separate routes

Reply:
The middleware checkAccountOwnership blocks on initializeRecoveryMethod. Should we refactor this on contract helper? The endpoints are tiny wrappers with their respective middleware to checkAccountOwnership or checkAccountDoesNotExist